src/Symfony/Component/Security/Http/AccessToken/RFC6750AuthenticationFailureHandler.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authentication\AuthenticationFailureHandlerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Provides an implementation of the RFC6750 of an authentication via
+ * an access token.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+class RFC6750AuthenticationFailureHandler implements AuthenticationFailureHandlerInterface
```

Comment:
This class is not wired by default?

Reply:
Indeed, related to https://github.com/symfony/symfony/pull/46428#discussion_r897303339. 
I second Wouter's suggestion there about making this the default failure handling + inlining it in the authenticator so that it is used when no `failure_handler` is configured.
Seems like we have no default authentication success handling either, could we find a sensible one to be inlined in the `onAuthenticationSuccess()` as well?

Reply:
At the moment, the default failure handler is set to `null`.
Since the method `supports` returns `null`, the `authenticate` method has to return `null` as well otherwise anonymous user access fails even if `PUBLIC_ACCESS` is present for the route. I may have missed something but I cannot set a handler and allow anonymous access at the same time.

The class `RFC6750AuthenticationFailureHandler` is not wired for now. The idea is to let users decide to use it or not.

```yaml
services:
    app.custom_failure_handler:
        class: Symfony\Component\Security\Http\AccessToken\RFC6750AuthenticationFailureHandler
        arguments:
            - 'My API'
        calls:
            - setTranslator: ['@translator']
```

Reply:
To be honest, this could be a more generic handler. I will try to find another way for this.

Reply:
> Since the method supports returns null, the authenticate method has to return null as well otherwise anonymous user access fails even if PUBLIC_ACCESS is present for the route. I may have missed something but I cannot set a handler and allow anonymous access at the same time.

`supports()` returning `null` should have no influence on the behavior of any other method in the authenticator. I'll try to find some time tomorrow to test this PR in a project, so I can have a look tomorrow about this issue with anonymous users.

I'm still leaning towards inlining this code in the authenticator. If someone wants to use the code in a different authenticator, we can always accept a PR moving it to a special failure handler in the future (the typical OSS tactic to reduce maintenance cost is to not implement the bare minimum, and let people contribute things if they want more).

Reply:
The test that is failing is [`testAnonymousAccessIsGranted`](https://github.com/symfony/symfony/pull/46428/files#diff-115580d4348ff16c78d09d377df33b8f9074867515f43bd9a116302eedcb706cR33-R42).
It is unable to reach `/ bar`; a 401 http error is returned. The [firewall configuration for this test is here](https://github.com/symfony/symfony/pull/46428/files#diff-733fda09dfc62622cb471f01e84876930ad05b9ec4f0834cb5e37d82443845d1R19-R29)

Reply:
Turns out the `support()` method was returning `null` in the wrong cases. It's used as a replacement for `true`, not `false`. I've fixed it in https://github.com/symfony/symfony/pull/46428/commits/ea9d2b31739152f37e908f261932932b44f32434

I've also pushed another commit to your branch, which merges the RFC failure handler into the authenticator, as suggested by @chalasr and me. Imho, this is now ready to go :)